python/ray/tune/progress_reporter.py

```diff
@@ -109,52 +115,89 @@ def trial_progress_str(trials, metrics=None, fmt="psql", max_rows=100):
         return delim.join(messages)
 
     num_trials = len(trials)
-    trials_per_state = {}
+    trials_by_state = collections.defaultdict(list)
     for t in trials:
-        trials_per_state[t.status] = trials_per_state.get(t.status, 0) + 1
+        trials_by_state[t.status].append(t)
+
+    num_trials_by_state = {s: len(trials_by_state[s]) for s in trials_by_state}
     messages.append("Number of trials: {} ({})".format(num_trials,
-                                                       trials_per_state))
+                                                       num_trials_by_state))
     for local_dir in sorted({t.local_dir for t in trials}):
         messages.append("Result logdir: {}".format(local_dir))
 
     if num_trials > max_rows:
-        overflow = num_trials - max_rows
         # TODO(ujvl): suggestion for users to view more rows.
         messages.append("Table truncated to {} rows ({} overflow).".format(
-            max_rows, overflow))
+            max_rows, num_trials - max_rows))
+        trials = _fair_filter_trials(trials_by_state, max_rows)
 
     # Pre-process trials to figure out what columns to show.
     keys = list(metrics or DEFAULT_PROGRESS_KEYS)
     keys = [k for k in keys if any(t.last_result.get(k) for t in trials)]
-
     # Build trial rows.
-    trial_table = []
     params = list(set().union(*[t.evaluated_params for t in trials]))
-    for trial in trials[:min(num_trials, max_rows)]:
-        trial_table.append(_get_trial_info(trial, params, keys))
+    trial_table = [_get_trial_info(trial, params, keys) for trial in trials]
     # Parse columns.
     parsed_columns = [REPORTED_REPRESENTATIONS.get(k, k) for k in keys]
     columns = ["Trial name", "status", "loc"]
     columns += params + parsed_columns
     messages.append(
         tabulate(trial_table, headers=columns, tablefmt=fmt, showindex=False))
+    return delim.join(messages)
 
-    # Build trial error rows.
+
+def trial_errors_str(trials, fmt="psql", max_rows=100):
+    """Returns a readable message regarding trial errors.
+
+    Args:
+        trials (List[Trial]): List of trials to get progress string for.
+        fmt (str): Output format (see tablefmt in tabulate API).
+        max_rows (int): Maximum number of rows in the error table.
+    """
+    messages = []
     failed = [t for t in trials if t.error_file]
-    if len(failed) > 0:
-        messages.append("Number of errored trials: {}".format(len(failed)))
+    num_failed = len(failed)
+    if num_failed > 0:
+        messages.append("Number of errored trials: {}".format(num_failed))
+        if num_failed > max_rows:
```

Comment:
Shouldn't it show how many are truncated for each type?

Reply:
this is at the table-level for both tables. i can do as you suggest in `trial_progress_str` though.